Add capabilities/platform documentation (#3542)
meilisearch/documentation: capabilities/platform/management/experimental_features.mdx

---
title: Experimental features
description: Enable or disable experimental Meilisearch features from the Cloud project settings.
---

Meilisearch occasionally ships features in an experimental state before promoting them to stable. Experimental features are fully functional but their API or behavior may change between Meilisearch versions without a deprecation period. Be careful when enabling them in production environments.

You can enable or disable experimental features per project from **Project Settings > General**.

<Frame>
  <img src="/assets/images/platform/management/experimental_features.png" alt="Experimental features section in Project Settings showing checkboxes for each available feature" />
</Frame>

Check the box next to a feature to enable it. Changes take effect immediately without a restart or re-index.

## Available features

> The list of available features evolves with each Meilisearch release and is not fixed. The features shown in the Cloud UI at any given time depend on your project's current Meilisearch version. As an example, the current UI shows:

Reviewer: ```suggestion
The list of available features evolves with each Meilisearch release and is not fixed. The features shown in the Cloud UI at any given time depend on your project's current Meilisearch version. In the example above, the UI shows:
```

Author: Applied the suggestion.

---

Meilisearch occasionally ships features in an experimental state before promoting them to stable. Experimental features are fully functional but their API or behavior may change between Meilisearch versions without a deprecation period. Be careful when enabling them in production environments.

You can enable or disable experimental features per project from **Project Settings > General**.

<Frame>
  <img src="/assets/images/platform/management/experimental_features.png" alt="Experimental features section in Project Settings showing checkboxes for each available feature" />
</Frame>

Check the box next to a feature to enable it. Changes take effect immediately without a restart or re-index.

## Available features

The list of available features evolves with each Meilisearch release and is not fixed. The features shown in the Cloud UI at any given time depend on your project's current Meilisearch version. As an example, the current UI shows:

| Feature | Description |
|---------|-------------|
| **Edit documents by function** | Apply custom transformations to documents using Rhai scripts, from simple formatting changes to complex logic |

> | **"CONTAINS" and "STARTS_WITH" filters** | Adds the `CONTAINS` filter operator to filter documents where an attribute includes a specific substring |

Author: Fixed: the row now documents both the CONTAINS and STARTS_WITH operators.

> | **Dynamic Search Rules** | Enable rules that change how search results are ordered when specific conditions are met (e.g. pinning) |

Reviewer 2: This is not there anymore since we're selling it

Author: Done: removed Dynamic Search Rules from the experimental features list.
